### ISSUE TITLE
cli: when interacting with a network output tx hash

### ISSUE THREAD

**Issue by tomerweller** [stellar/stellar-cli#816]
Maybe even with a link to the transaction page in a supported block explorer

**Comment by leighmcculloch** [comment]
It would be helpful if in general the CLI outputted the tx hash for any tx it has built for any purpose.

**Comment by leighmcculloch** [comment]
This feature missing is very inconvenient when wanting to introspect further into a transaction the CLI has submitted.

**Comment by leighmcculloch** [comment]
I've opened a separate issue for being able to get the tx hash for any tx:
- https://github.com/stellar/soroban-cli/issues/1316

It doesn't compete with this feature exactly, but I'd also find it useful and it's semi-related.

**Comment by janewang** [comment]
Can log without waiting for the tx to complete

**Comment by fnando** [comment]
I thought about logging the transaction hash as a debug logging, which would then be available when using `--very-verbose`. Wdyt?

```console
$ stellar --very-verbose contract deploy --wasm target/wasm32-unknown-unknown/release/hello_world.wasm --source alice --network testnet --alias hello
2024-07-16T16:51:08.633869Z DEBUG soroban_cli::commands::contract::deploy::wasm: transaction hash: b92b5cd32a13eb40400b4a71361d256074cdb381df554342284ceda38202039e
CB4YDNXFAZ42EM2DJG2PXTIVXXAKTKUYOCFJ2IGVTG5FFYLNS3HV7PMO
```

I don't love that the verbose logging includes that much noise, but it seems like a good choice and backwards compatible.

**Comment by willemneal** [comment]
Debug is verbose. Trace is very verbose

**Comment by leighmcculloch** [comment]
I think the transaction hash could be info.

**Comment by leighmcculloch** [comment]
Our commands could be more communicative about what they're doing, and not in terms of logs, but in terms of stderr output so that the user understands what is happening. We've started to do this on some commands, but not most, where we have informational output about each step.

Something like:

```
$ stellar contract deploy --wasm out/hello_world.wasm --source alice --network testnet --alias hello
ℹ️ Simulating install transaction...
🌎 Submitting install transaction... b92b5cd32a13eb40400b4a71361d256074cdb381df554342284ceda38202039e
ℹ️ Simulating deploy transaction...
🌎 Submitting deploy transaction... b92b5cd32a13eb40400b4a71361d256074cdb381df554342284ceda38202039e
✅ Deployed
CB4YDNXFAZ42EM2DJG2PXTIVXXAKTKUYOCFJ2IGVTG5FFYLNS3HV7PMO
```

**Comment by fnando** [comment]
@leighmcculloch I like that. Should we send everything to stderr and just the contract id to stdout, so it's backward compatible?

**Comment by willemneal** [comment]
Yeah definitely. Stdout should just be for the Contract Id so that it can be captured in scripts.

**Comment by leighmcculloch** [comment]
+1 Ideally every command that could be used in a scriptable way has a single entity on stdout for scripting.

**Comment by leighmcculloch** [comment]
It could be beneficial to do a quick sketch in a google doc for what each of the major commands are, and what type of stderr output we'd like for them. Taking a holistic view of how this should play out. Then we can make sure we have a good unified and consistent story for what this UX would look like.

**Comment by fnando** [comment]
I think we can start with `stellar contract deploy`, laying out the foundation with functions like `info(message)`, `link(url)`, and so on, which would know which emoji to use and any other annotations we'd like to add.
